src/Dom/Document.php

```diff
@@ -1021,6 +1023,34 @@ private function restore_doctype_node( $html ) {
 		return preg_replace( self::HTML_RESTORE_DOCTYPE_PATTERN, '\1!\3\4>', $html, 1 );
 	}
 
+	/**
+	 * Normalizes HTML attributes to be HTML5 compatible.
+	 *
+	 * Conditionally removes html[xmlns], and converts html[xml:lang] to html[lang].
+	 */
+	private function normalize_html_attributes() {
+		$html = $this->getElementsByTagName( 'html' )->item( 0 );
+		if ( ! $html->hasAttributes()  ) {
+			return;
+		}
+
+		$xmlns                = $html->attributes->getNamedItem( 'xmlns' );
+		$xmlns_value_to_strip = 'http://www.w3.org/1999/xhtml';
+		if ( $xmlns && $xmlns_value_to_strip === $xmlns->nodeValue ) {
+			$html->removeAttributeNode( $xmlns );
+		}
+
+		$xml_lang = $html->attributes->getNamedItem( 'xml:lang' );
```

Comment:
Have you considered [`hasAttributeNS`](https://www.php.net/manual/en/domelement.hasattributens.php) or [`getAttributeNS`](https://www.php.net/manual/en/domelement.getattributens.php)? 

It seems these don't work either. I would have thought at least that this would work: 

```php
$dom->documentElement->getAttributeNodeNS( 'http://www.w3.org/XML/1998/namespace', 'lang' );
```

But it doesn't. Perhaps it's due to `loadHTML` and that the document is not XML in the first place.

Using `$dom->documentElement->attributes->getNamedItem( 'xml:lang' )` works in all PHP versions 5+. So seems like a good one to use.

Reply:
Nice, thanks for looking at this.